system-addon/lib/ActivityStreamStorage.jsm

```diff
@@ -65,6 +65,27 @@ this.ActivityStreamStorage = class ActivityStreamStorage {
     });
   }
 
+  /**
+   * createOrOpenDb - Open a db (with this.dbName) if it exists.
+   *                  If it does not exist, create it.
+   *                  If an error occurs, deleted the db and attempt to
+   *                  re-create it.
+   * @returns Promise that resolves with a db instance
+   */
+  async createOrOpenDb() {
+    let db;
+    try {
+      db = await this._openDatabase();
```

Comment:
nit: this can just `return await this._openDatabase();` and similar below without needing `let db;`